doc/v5.0/man3/MPI_Comm_accept.3
.\" Automatically generated by Pandoc 2.14.2
.\"
.TH "MPI_Comm_accept" "3" "" "2021-08-30" "Open MPI"
.hy
.SH Name
.PP
\f[C]MPI_Comm_accept\f[R] - Establishes communication with a client.
.SH Syntax
.SS C Syntax
.IP
.nf
\f[C]
#include <mpi.h>

int MPI_Comm_accept(const char *port_name, MPI_Info info, int root, MPI_Comm comm, MPI_Comm *newcomm)
\f[R]
.fi
.SS Fortran Syntax
.IP
.nf
\f[C]
USE MPI
! or the older form: INCLUDE \[aq]mpif.h\[aq]

MPI_COMM_ACCEPT(PORT_NAME, INFO, ROOT, COMM, NEWCOMM, IERROR)
    CHARACTER*(*)   PORT_NAME
    INTEGER     INFO, ROOT, COMM, NEWCOMM, IERROR
\f[R]
.fi
.SS Fortran 2008 Syntax
.IP
.nf
\f[C]
USE mpi_f08

MPI_Comm_accept(port_name, info, root, comm, newcomm, ierror)
    CHARACTER(LEN=*), INTENT(IN) :: port_name
    TYPE(MPI_Info), INTENT(IN) :: info
    INTEGER, INTENT(IN) :: root
    TYPE(MPI_Comm), INTENT(IN) :: comm
    TYPE(MPI_Comm), INTENT(OUT) :: newcomm
    INTEGER, OPTIONAL, INTENT(OUT) :: ierror
\f[R]
.fi
.SH Input Parameters
.IP \[bu] 2
\f[C]port_name\f[R] : Port name (string, used only on \f[I]root\f[R]).
.IP \[bu] 2
\f[C]info\f[R] : Options given by root for the accept (handle, used only
on root).
No options currently supported.
.IP \[bu] 2
\f[C]root\f[R] : Rank in \f[I]comm\f[R] of root node (integer).
.IP \[bu] 2
\f[C]comm\f[R] : Intracommunicator over which call is collective
(handle).
.SH Output Parameters
.IP \[bu] 2
\f[C]newcomm\f[R] : Intercommunicator with client as remote group
(handle)
.IP \[bu] 2
\f[C]IERROR\f[R] : Fortran only: Error status (integer).
.SH Description
.PP
\f[C]MPI_Comm_accept\f[R] establishes communication with a client.
It is collective over the calling communicator.
It returns an intercommunicator that allows communication with the
client, after the client has connected with the
\f[C]MPI_Comm_accept\f[R] function using the \f[C]MPI_Comm_connect\f[R]
function.
The \f[C]port_name\f[R] must have been established through a call to
\f[C]MPI_Open_port\f[R] on the \f[C]root\f[R].
.SH Errors
.PP
Almost all MPI routines return an error value; C routines as the value
of the function and Fortran routines in the last argument.
Before the error value is returned, the current MPI error handler is
called.
By default, this error handler aborts the MPI job, except for I/O
function errors.
The error handler may be changed with \f[C]MPI_Comm_set_errhandler\f[R];
the predefined error handler \f[C]MPI_ERRORS_RETURN\f[R] may be used to
cause error values to be returned.
See the MPI man page for a full list of MPI error codes.
.SH See Also
.PP
\f[C]MPI_Comm_connect\f[R](3) \f[C]MPI_Open_port\f[R](3)


doc/v1.4/man3/MPI_Comm_accept.3.php
<?php
$topdir = "../../..";
$title = "MPI_Comm_accept(3) man page (version 1.4.5)";
$meta_desc = "Open MPI v1.4.5 man page: MPI_COMM_ACCEPT(3)";

include_once("$topdir/doc/nav.inc");
include_once("$topdir/includes/header.inc");
?>
<p> <a href="../">&laquo; Return to documentation listing</a></p>
   <!-- manual page source format generated by PolyglotMan v3.2, -->
<!-- available at http://polyglotman.sourceforge.net/ -->

<body bgcolor='white'>
<a href='#toc'>Table of Contents</a><p>

<h2><a name='sect0' href='#toc0'>Name</a></h2>
<b>MPI_Comm_accept </b> - Establishes communication with a client.
<p>
<h2><a name='sect1' href='#toc1'>Syntax</a></h2>

<h2><a name='sect2' href='#toc2'>C
Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
int MPI_Comm_accept(char *port_name, MPI_Info info, int root, MPI_Comm
comm, MPI_Comm *newcomm)
</pre>
<h2><a name='sect3' href='#toc3'>Fortran Syntax</a></h2>
<br>
<pre>INCLUDE &rsquo;mpif.h&rsquo;
MPI_COMM_ACCEPT(PORT_NAME, INFO, ROOT, COMM, NEWCOMM, IERROR)
<tt> </tt>&nbsp;<tt> </tt>&nbsp;CHARACTER*(*)<tt> </tt>&nbsp;<tt> </tt>&nbsp;PORT_NAME
<tt> </tt>&nbsp;<tt> </tt>&nbsp;INTEGER<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;<tt> </tt>&nbsp;INFO, ROOT, COMM, NEWCOMM, IERROR
</pre>
<h2><a name='sect4' href='#toc4'>C++ Syntax</a></h2>
<br>
<pre>#include &lt;mpi.h&gt;
MPI::Intercomm MPI::Intracomm::Accept(const char* port_name,
<tt> </tt>&nbsp;<tt> </tt>&nbsp;const MPI::Info&amp; info, int root) const
</pre>
<h2><a name='sect5' href='#toc5'>Input Parameters</a></h2>

<dl>

<dt>port_name </dt>
<dd>Port name (string, used only on <i>root</i>). </dd>

<dt>info </dt>
<dd>Options
given by root for the accept (handle, used only on root). No options currently
supported.  </dd>

<dt>root </dt>
<dd>Rank in <i>comm</i> of root node (integer). </dd>

<dt>comm </dt>
<dd>Intracommunicator
over which call is collective (handle).
<p> </dd>
</dl>

<h2><a name='sect6' href='#toc6'>Output Parameters</a></h2>

<dl>

<dt>newcomm </dt>
<dd>Intercommunicator
with client as remote group (handle) </dd>

<dt>IERROR </dt>
<dd>Fortran only: Error status
(integer).
<p> </dd>
</dl>

<h2><a name='sect7' href='#toc7'>Description</a></h2>
MPI_Comm_accept establishes communication with a
client. It is collective over the calling communicator. It returns an intercommunicator
that allows communication with the client, after the client has connected
with the MPI_Comm_accept function using the <a href="../man3/MPI_Comm_connect.3.php">MPI_Comm_connect</a> function. <p>
The
<i>port_name</i> must have been established through a call to <a href="../man3/MPI_Open_port.3.php">MPI_Open_port</a> on
the root.
<p>
<p>
<h2><a name='sect8' href='#toc8'>Errors</a></h2>
Almost all MPI routines return an error value; C routines
as the value of the function and Fortran routines in the last argument.
C++ functions do not return errors. If the default error handler is set
to MPI::ERRORS_THROW_EXCEPTIONS, then on error the C++ exception mechanism
will be used to throw an MPI:Exception object. <p>
Before the error value is
returned, the current MPI error handler is called. By default, this error
handler aborts the MPI job, except for I/O function errors. The error handler
may be changed with <a href="../man3/MPI_Comm_set_errhandler.3.php">MPI_Comm_set_errhandler</a>; the predefined error handler
MPI_ERRORS_RETURN may be used to cause error values to be returned.   <p>
See
the MPI man page for a full list of MPI error codes.
<p>
<h2><a name='sect9' href='#toc9'>See Also</a></h2>
<a href="../man3/MPI_Comm_connect.3.php">MPI_Comm_connect</a>
<a href="../man3/MPI_Open_port.3.php">MPI_Open_port</a> <br>

<p> <p>

<hr><p>
<a name='toc'><b>Table of Contents</b></a><p>
<ul>
<li><a name='toc0' href='#sect0'>Name</a></li>
<li><a name='toc1' href='#sect1'>Syntax</a></li>
<li><a name='toc2' href='#sect2'>C Syntax</a></li>
<li><a name='toc3' href='#sect3'>Fortran Syntax</a></li>
<li><a name='toc4' href='#sect4'>C++ Syntax</a></li>
<li><a name='toc5' href='#sect5'>Input Parameters</a></li>
<li><a name='toc6' href='#sect6'>Output Parameters</a></li>
<li><a name='toc7' href='#sect7'>Description</a></li>
<li><a name='toc8' href='#sect8'>Errors</a></li>
<li><a name='toc9' href='#sect9'>See Also</a></li>
</ul>


<p> <a href="../">&laquo; Return to documentation listing</a></p>
<?php
include_once("$topdir/includes/footer.inc");
